sonarts-core/src/rules/noStatementsSameLineRule.ts

```diff
@@ -0,0 +1,137 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as tslint from "tslint";
+import * as ts from "typescript";
+import { SonarRuleMetaData } from "../sonarRule";
+import { SonarRuleVisitor } from "../utils/sonarAnalysis";
+import {
+  isIfStatement,
+  isBlock,
+  isIterationStatement,
+  isFunctionExpression,
+  isArrowFunction,
+  is,
+} from "../utils/nodes";
+import { startLineAndCharacter, endLineAndCharacter } from "../utils/navigation";
+
+export class Rule extends tslint.Rules.AbstractRule {
+  public static metadata: SonarRuleMetaData = {
+    ruleName: "no-statements-same-line",
+    description: "Statements should be on separate lines",
+    rationale: tslint.Utils.dedent``,
+    optionsDescription: "",
+    options: null,
+    rspecKey: "RSPEC-122",
+    type: "maintainability",
+    typescriptOnly: false,
+  };
+
+  public static MESSAGE = "Reformat the code to have only one statement per line.";
+
+  public apply(sourceFile: ts.SourceFile): tslint.RuleFailure[] {
+    const visitor = new Visitor(this.getOptions().ruleName).visit(sourceFile);
+
+    visitor.statementsPerLine.forEach(statements => {
+      if (statements.length > 1) {
+        const issue = visitor.addIssue(statements[1], Rule.MESSAGE);
+        statements.slice(2).forEach(statement => issue.addSecondaryLocation(statement));
+      }
+    });
+
+    return visitor.getIssues();
+  }
+}
+
+class Visitor extends SonarRuleVisitor {
+  statementsPerLine: Map<number, ts.Statement[]> = new Map();
+  excludedStatements: ts.Statement[] = [];
+
+  private static readonly STATEMENT_KINDS = [
+    ts.SyntaxKind.VariableStatement,
+    ts.SyntaxKind.ExpressionStatement,
+    ts.SyntaxKind.IfStatement,
+    ts.SyntaxKind.DoStatement,
+    ts.SyntaxKind.WhileStatement,
+    ts.SyntaxKind.ForStatement,
+    ts.SyntaxKind.ForInStatement,
+    ts.SyntaxKind.ForOfStatement,
+    ts.SyntaxKind.ContinueStatement,
+    ts.SyntaxKind.BreakStatement,
+    ts.SyntaxKind.ReturnStatement,
+    ts.SyntaxKind.WithStatement,
+    ts.SyntaxKind.SwitchStatement,
+    ts.SyntaxKind.ThrowStatement,
+    ts.SyntaxKind.TryStatement,
+    ts.SyntaxKind.DebuggerStatement,
+  ];
+
+  public visitNode(node: ts.Node) {
+    if (isIfStatement(node)) {
+      this.checkForExcludedStatement(node.thenStatement, node);
+    }
+
+    if (isIterationStatement(node)) {
+      this.checkForExcludedStatement(node.statement, node);
+    }
+
+    if (isFunctionExpression(node) || isArrowFunction(node)) {
+      this.checkFunctionException(node);
+    }
+
+    if (is(node, ...Visitor.STATEMENT_KINDS) && !this.excludedStatements.includes(node as ts.Statement)) {
+      const line = startLineAndCharacter(node).line;
+      const statementsOnThisLine = this.statementsPerLine.get(line) || [];
+      statementsOnThisLine.push(node as ts.Statement);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS4325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=issue-489&issues=AWGz_Gi-UhrvXJpox_ld&open=AWGz_Gi-UhrvXJpox_ld)

[ISSUE_KEY]: AWGz_Gi-UhrvXJpox_ld